core/src/main/java/com/datastax/oss/driver/internal/core/session/ReprepareOnUp.java

```diff
@@ -224,37 +226,51 @@ private void startWorker() {
               new Prepare(
                   payload.query, (payload.keyspace == null ? null : payload.keyspace.asInternal())),
               payload.customPayload)
-          .handle(
+          .handleAsync(
               (result, error) -> {
                 // Don't log, AdminRequestHandler does already
                 startWorker();
                 return null;
-              });
+              },
+              adminExecutor);
     }
   }
 
   @VisibleForTesting
   protected CompletionStage<AdminResult> queryAsync(
       Message message, Map<String, ByteBuffer> customPayload, String debugString) {
-    ThrottledAdminRequestHandler<AdminResult> reprepareHandler =
-        ThrottledAdminRequestHandler.query(
-            channel,
-            message,
-            customPayload,
-            timeout,
-            throttler,
-            metricUpdater,
-            logPrefix,
-            debugString);
-    return reprepareHandler.start();
+    DriverChannel channel = pool.next();
+    if (channel == null) {
+      return CompletableFutures.failedFuture(
+          new BusyConnectionException("Found no channel to execute reprepare query"));
+    } else {
+      ThrottledAdminRequestHandler<AdminResult> reprepareHandler =
+          ThrottledAdminRequestHandler.query(
+              channel,
+              false,
+              message,
+              customPayload,
+              timeout,
+              throttler,
+              metricUpdater,
+              logPrefix,
+              debugString);
+      return reprepareHandler.start();
+    }
   }
 
   @VisibleForTesting
   protected CompletionStage<ByteBuffer> prepareAsync(
       Message message, Map<String, ByteBuffer> customPayload) {
-    ThrottledAdminRequestHandler<ByteBuffer> reprepareHandler =
-        ThrottledAdminRequestHandler.prepare(
-            channel, message, customPayload, timeout, throttler, metricUpdater, logPrefix);
-    return reprepareHandler.start();
+    DriverChannel channel = pool.next();
```

Comment:
Side-effect of reacquiring every time: the reprepare requests will now be distributed across pooled connections, vs using just 1 in the previous implementation. This is an improvement.